sympy/functions/elementary/tests/test_piecewise.py

```diff
@@ -293,9 +293,9 @@ def test_piecewise_solve():
 
     g = Piecewise(((x - 5)**5, x >= 2), (f, x < 2))
     assert solve(g, x) == [5]
-
+    # solve returing {2,5} for below.
     g = Piecewise(((x - 5)**5, x >= 2), (f, True))
-    assert solve(g, x) == [5]
```

Comment:
you should modify this test as necessary but not change teh function from solve to solveset. In the test_solveset suite is where you should add the new tests.